### PR TITLE
Feat/secret mgmt

### PR DESCRIPTION
**Pull request by yannickstruyf3**
**What this PR does / why we need it**:
- Change the way credentials are handled:
     -  Allow credentials to be defined on nutanixCluster and Manager
     - Add Prism Central attribute on nutanixCluster
     - Remove endpoint, port flags from secret
     - Implement optional insecure mode
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
N/A
**How Has This Been Tested?**:
- Create CAPI cluster with new cluster-template.yml file
- Use different user/pass at manager and cluster level and check VM creation
- Use insecure false on PC with invalid certs and check failure

**Special notes for your reviewer**:

**Release note**:
```release-note
- Change the way credentials are handled:
     - Allow credentials to be defined on nutanixCluster and Manager
     - Add Prism Central attribute on nutanixCluster
     - Remove endpoint, port flags from secret
     - Implement optional insecure mode
```

### CODE REVIEW

**Review by tuxtof — DISMISSED** [review]

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -20,8 +20,10 @@ rules:
   resources:
   - secrets
   verbs:
+  - delete
```
Comment (deepakm-ntnx) [inline config/rbac/role.yaml:23]: why is this needed?

**Review by deepakm-ntnx — APPROVED** [review]

File: config/rbac/role.yaml (hunk 1 above)
Comment (yannickstruyf3) [reply config/rbac/role.yaml:23]: @deepakm-ntnx The secret is created by the cluster-template. When the nutanixcluster object is created, it will add an owner ref to the secret and also remove the secret if nutanixcluster object is deleted. In order to delete the owned secret this permission is required.

**Review by tuxtof — APPROVED** [review]
